presto-tests/src/main/java/com/facebook/presto/tests/statistics/MetricComparison.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tests.statistics;
+
+import com.facebook.presto.sql.planner.plan.PlanNode;
+
+import java.util.Optional;
+
+import static com.facebook.presto.tests.statistics.MetricComparison.Result.DIFFER;
+import static com.facebook.presto.tests.statistics.MetricComparison.Result.MATCH;
+import static com.facebook.presto.tests.statistics.MetricComparison.Result.NO_BASELINE;
+import static com.facebook.presto.tests.statistics.MetricComparison.Result.NO_ESTIMATE;
+import static java.lang.String.format;
+
+public class MetricComparison
```

Comment:
> we always create the list of MetricComparisons and immediately compute their result

You're correct in that, but the the result is needed as an object (rather than an immediate console output) anyway. We need both the kind of result (`MetricComparison.Result`) and the node and the `Metric` to which that result applies. We need this data for two distinct purposes: to summarize & print it (`testCostEstimatesVsRealityDifferences`), and to assert upon it (`testEstimateForSimpleQuery`, which is expanded in @kokosing's work). Hence the `MetricComparison` value object.

If you mean the `result()` method is unnecessary, or the `estimateMatchesReality` is not needed - those are extension points for the future where tests specify their error tolerance themselves. We need the tolerance to know whether the result is `MATCH` or `NO_MATCH`, so - since the tolerance will be injected - we can not precompute the `Result`. This already happened in @kokosing's [commit](https://github.com/Teradata/presto/commit/084f6b97adae308984c202c688b0721607335e62#diff-e42dc47ae364fc9e486762bf93251329R58 ) which enabled the metric tests to [look like this](https://github.com/Teradata/presto/commit/084f6b97adae308984c202c688b0721607335e62#diff-44e67b7c369c05e242947b521af424c4R46)